flow-typed/redux.js

```diff
@@ -0,0 +1,74 @@
+declare module 'redux' {
+
+  /*
+
+    S = State
+    A = Action
+
+  */
+
+  declare type Dispatch<S, A> = (action: A) => A;
```

Comment:
I think this definition is a bit more rigid than the middleware API allows. For example, `redux-actions` would allow a `Dispatch` object to take a param `action` of type `{type: T, payload: Promise<P>}`, but would then return `Promise<{type: T, payload: P}>`.